### ISSUE TITLE
Transit key usage count limitation 

### ISSUE THREAD

**Issue by evan361425** [hashicorp/vault#13025]
**Is your feature request related to a problem? Please describe.**
Key needs to be [rotate after 2^32 encryptions](https://www.vaultproject.io/docs/secrets/transit#nist-rotation-guidance). How do I know the count?

**Describe the solution you'd like**
- [Read key configuration](https://www.vaultproject.io/api-docs/secret/transit#read-key) should show
- Telemetry expose status

**Describe alternatives you've considered**
Proxy stats but it have limitation if using TLS on Vault.

**Explain any additional use-cases**

**Additional context**

**Comment by hsimon-hashicorp** [comment]
Hi @evan361425 - this is something that we would like to implement, but there's implications as mentioned by @ixe013 and it's a bit challenging to do with the existing plugin system. I will continue to track this for future feedback to the team.

**Comment by evan361425** [comment]
@ixe013 It is acceptable to loss some correctness when dealing with 2^32 counter. Maybe active node periodically summarize all standby nodes Transit counter is a solution.

I've search for 'distributed counter' and 'replica MVCC'(it also needs an algorithm to handling distributed counter in DB), but then realized we no need that high correctness.

**Comment by ixe013** [comment]
Indeed, precision is not required. The fact that you even worry about key rotation puts you in a league appart.

AppRole already have a concept where they [store secret ids locally (in the cluster)](https://www.vaultproject.io/api/auth/approle#local_secret_ids). They are not replicated at all from a performance replica down to the primary cluster. Using the same trick, we would at least avoid cross-region or cross-cloud replication.

I also remember working with an LDAP server that had a concept of "bus stop replication". Non-urgent stuff like this count would wait for the next important replication message and go along for the ride. But it would require logic on the receiving end to add the replicated count to the primary's. Not easy.

If we make the count feature opt-in we might have something. For example, count the usage only if one of the following are set:
 - `expected_number_of_uses` The number of operations allowed before this key wears out
 - `number_of_uses_warning_threshold` The number of operations before a warning about key wear is returned

(ps I am not an Hashicorp employee, I have no power on what gets in or not. I'm just looking for [another](https://github.com/hashicorp/vault/pull/11000) small fun project to do.)

**Comment by hsimon-hashicorp** [comment]
Hi everyone! Just a note that we are tracking this request internally, and have begun the process of working it into a future release. I can't make any guarantees about whether or when this gets implemented as requested, but our product and engineering teams are on it. Thanks a bunch! :)

**Comment by DemiMarie** [comment]
Another alternative would be to maintain a stateful counter on each node, with each node having 64-bit nonce + 32-bit node ID.  The node ID would need to be managed by Vault, though, which would be tricky.
